Core/Python/set_power_state.py

```diff
@@ -186,18 +186,25 @@ def get_execution_detail(job_hist_resp, headers, job_hist_url):
 def get_device_list(ip_address, headers):
 	""" Get list of devices from OME """
 	ome_device_list = None
-	device_url = 'https://%s/api/DeviceService/Devices' % ip_address
-	device_response = requests.get(device_url, headers=headers, verify=False)
-	if device_response.status_code == 200:
-		dev_json_response = device_response.json()
-		if dev_json_response['@odata.count'] > 0:
-			ome_device_list = [x['Id'] for x in dev_json_response['value']]
+	next_link_url = 'https://%s/api/DeviceService/Devices' % ip_address
+	while next_link_url is not None:
+		device_response = requests.get(next_link_url, headers=headers, verify=False)
+		next_link_url = None
+		if device_response.status_code == 200:
+			dev_json_response = device_response.json()
+			if dev_json_response['@odata.count'] <= 0:
+				print("No devices found at ", ip_address)
+				return
+
+			if '@odata.nextLink' in dev_json_response:
+				next_link_url = 'https://%s/' %ip_address + dev_json_response['@odata.nextLink']
+
+			if dev_json_response['@odata.count'] > 0:
+				ome_device_list = [x['Id'] for x in dev_json_response['value']]
```

Comment:
This might collect the devices only from the last call. Think we should collect the data from all responses in the loop and then finally collect all the ids.